platform/android/app/src/main/kotlin/coop/polypoly/polypod/features/Feature.kt

```diff
@@ -6,25 +6,36 @@ import android.graphics.Color
 import android.util.DisplayMetrics
 import java.util.zip.ZipFile
 
+interface FeatureInterface {
```

Comment:
Yeah, the constructor thing does not work. A second constructor needs to reference the first (primary) one, and `findUrl` requires a manifest, which I don't want to reconstruct or pass to the second constructor, since it is such a pain to deal with.

Reply:
Not a huge fan of having something with `Interface` in the name either - maybe the interface could be `Feature` and the concrete implementation below `ZipFeature` or something? Anything _other_ than adding stuff like `Interface` or `Impl` (yuck) seems better :D

Reply:
> Yeah, the constructor thing does not work. A second constructor needs to reference the first (primary) one, and `findUrl` requires a manifest, which I don't want to reconstruct or pass to the second constructor, since it is such a pain to deal with.

You can have the constructor which initialises all the fields to be the primary constructor, and the current constructor will be secondary one. Also from the code, `findURL` just uses `Manifest` to access `links`, which could be sent at initialisation.